engine/test-src/org/pentaho/di/trans/steps/joinrows/JoinRowsTest.java

```diff
@@ -0,0 +1,83 @@
+/*! ******************************************************************************
+ *
+ * Pentaho Data Integration
+ *
+ * Copyright (C) 2002-2016 by Pentaho : http://www.pentaho.com
+ *
+ *******************************************************************************
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ******************************************************************************/
+
+package org.pentaho.di.trans.steps.joinrows;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.pentaho.di.core.logging.LogChannelInterface;
+import org.pentaho.di.trans.Trans;
+import org.pentaho.di.trans.TransMeta;
+import org.pentaho.di.trans.step.StepMeta;
+import org.pentaho.di.trans.step.StepMetaInterface;
+
+import java.io.File;
+
+/**
+ * @author Denis Mashukov
+ */
+public class JoinRowsTest {
+
+  private StepMetaInterface meta;
+  private JoinRowsData data;
+
+  @Before
+  public void setUp() throws Exception {
+    meta = new JoinRowsMeta();
+    data = new JoinRowsData();
+  }
+
+  @After
+  public void tearDown() {
+    meta = null;
+    data = null;
+  }
+
+  @Test
+  public void dataFileIsNull() throws Exception {
+    getJoinRows().dispose( meta, data );
+  }
+
+  @Test
+  public void disposeDataFiles() throws Exception {
+    data.file = new File[]{ Mockito.mock( File.class ), Mockito.mock( File.class ) };
+    getJoinRows().dispose( meta, data );
+  }
```

Comment:
@denis8703  Would be good to actually verify that something happens when the .dispose call is made-- i.e. that the files are deleted.

And with the dataFileIsNull() test above, you're really just testing that no NPE is thrown.  So why not check for that explicitly?  And maybe name your test something that reflects what it's testing for, or add a comment?